src/Servers/HttpSys/test/FunctionalTests/DelegateTests.cs

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Testing;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Server.HttpSys.FunctionalTests
+{
+    public class DelegateTests
+    {
+        private static readonly string _expectedResponseString = "Hello from delegatee";
+
+        [ConditionalFact]
+        [DelegateSupportedCondition(true)]
+        public async Task DelegateRequestTest()
+        {
+            var queueName = Guid.NewGuid().ToString();
+             using var receiver = Utilities.CreateHttpServer(out var receiverAddress, async httpContext =>
+            {
+                await httpContext.Response.WriteAsync(_expectedResponseString);
+            },
+            options =>
+            {
+                options.RequestQueueName = queueName;
+            });
+
+            DelegationRule wrapper = default;
+
+            using var delegator = Utilities.CreateHttpServer(out var delegatorAddress, httpContext =>
+            {
+                var transferFeature = httpContext.Features.Get<IHttpSysRequestTransferFeature>();
+                transferFeature.TransferRequest(wrapper);
+                return Task.FromResult(0);
+            });
+
+            var delegationProperty = delegator.Features.Get<IServerDelegationFeature>();
+            wrapper = delegationProperty.CreateDelegationRule(queueName, receiverAddress);
+
+            var responseString = await SendRequestAsync(delegatorAddress);
+            Assert.Equal(_expectedResponseString, responseString);
+            wrapper?.Dispose();
+        }
+
+        [ConditionalFact]
+        [DelegateSupportedCondition(true)]
+        public async Task DelegateAfterWriteToResponseBodyShouldThrowTest()
+        {
+            var queueName = Guid.NewGuid().ToString();
+            using var receiver = Utilities.CreateHttpServer(out var receiverAddress, async httpContext =>
+            {
+                await httpContext.Response.WriteAsync(_expectedResponseString);
+            },
```

Comment:
Avoid the false positive if this is invoked.
```suggestion
            using var receiver = Utilities.CreateHttpServer(out var receiverAddress, httpContext =>
            {
                httpContext.Response.StatusCode = StatusCodes.Status418ImATeapot;
                return Task.CompletedTask;
            },
```